File: ruby/find_first_duplicate.rb
def find_first_duplicate(arr)
  # type your code in here
  duplicate = []

  arr.each do |value|
    return value if duplicate.include?(value)

    duplicate.push(value)
  end

  -1
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 3"
  puts "=>", find_first_duplicate([2, 1, 3, 3, 2])

  puts "Expecting 5"
  puts "=>", find_first_duplicate([2, 8, 5, 1, 5, 3, 2, 5])

  puts "Expecting 7"
  puts "=>", find_first_duplicate([2, 7, 4, 7, 8, 5, 1, 5, 3, 2, 5])

  puts "Expecting: -1"
  puts "=>", find_first_duplicate([1, 2, 3, 4])

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/find_first_duplicate.js
function findFirstDuplicate(arr) {
  // type your code here
  const values = []

  for(const element of arr) {
    if (values.includes(element)) return element

    values.push(element)
  }

  return -1
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 3");
  console.log("=>", findFirstDuplicate([2, 1, 3, 3, 2]));

  console.log("Expecting 5");
  console.log("=>", findFirstDuplicate([2, 8, 5, 1, 5, 3, 2, 5]))

  console.log("Expecting 7");
  console.log("=>", findFirstDuplicate([2, 7, 4, 7, 8, 5, 1, 5, 3, 2, 5]))

  console.log("Expecting: -1");
  console.log("=>", findFirstDuplicate([1, 2, 3, 4]));
}

module.exports = findFirstDuplicate;

// Please add your pseudocode to this file

//Declare empty array
//Iterate over array
//Check if array contains value
// if so, return value
// if not, add value to array
//return -1 if no value is returned

// And a written explanation of your solution
